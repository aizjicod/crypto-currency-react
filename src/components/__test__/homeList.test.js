import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/createSttore';
import HomeList from '../homeList';

test('testing homeList component', () => {
  const component = render(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <HomeList
            name="etherium"
            img="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
            id="bitcoin"
            value={150}
            text="market"
          />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  );
  expect(component).toMatchSnapshot();
});

test('testing second homeList component', () => {
  const component = render(
    <React.StrictMode>

      <Provider store={store}>
        <BrowserRouter>
          <HomeList
            name="cordana"
            img="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
            id="cord"
            value={11357320}
            text="price"
          />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  );
  expect(component).toMatchSnapshot();
});
