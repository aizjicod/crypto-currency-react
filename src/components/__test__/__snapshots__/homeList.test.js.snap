// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing homeList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="home-list"
      >
        <div
          class="list-header"
        >
          <img
            alt="etherium"
            src="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
          />
          <h2>
            etherium
          </h2>
          <p
            class="list-p"
          >
            market: 150$
          </p>
        </div>
        <a
          class="list-nav"
          href="/details"
        >
          <svg
            class="more-arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
            />
          </svg>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="home-list"
    >
      <div
        class="list-header"
      >
        <img
          alt="etherium"
          src="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
        />
        <h2>
          etherium
        </h2>
        <p
          class="list-p"
        >
          market: 150$
        </p>
      </div>
      <a
        class="list-nav"
        href="/details"
      >
        <svg
          class="more-arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
          />
        </svg>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing second homeList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="home-list"
      >
        <div
          class="list-header"
        >
          <img
            alt="cordana"
            src="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
          />
          <h2>
            cordana
          </h2>
          <p
            class="list-p"
          >
            price: 11357320$
          </p>
        </div>
        <a
          class="list-nav"
          href="/details"
        >
          <svg
            class="more-arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
            />
          </svg>
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="home-list"
    >
      <div
        class="list-header"
      >
        <img
          alt="cordana"
          src="https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579"
        />
        <h2>
          cordana
        </h2>
        <p
          class="list-p"
        >
          price: 11357320$
        </p>
      </div>
      <a
        class="list-nav"
        href="/details"
      >
        <svg
          class="more-arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.536 21.886a1.004 1.004 0 0 0 1.033-.064l13-9a1 1 0 0 0 0-1.644l-13-9A.998.998 0 0 0 5 3v18a1 1 0 0 0 .536.886zM7 4.909 17.243 12 7 19.091V4.909z"
          />
        </svg>
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
